lib/pdf_fill/hash_converter.rb

```diff
@@ -102,6 +102,8 @@ def set_value(v, key_data, i, from_array_overflow = false, top_level_key = nil)
         add_to_extras(key_data, new_value, i, top_level_key)
 
         new_value = EXTRAS_TEXT
+      else
+        add_to_extras(key_data, new_value, i, top_level_key, overflow: false)
```

Comment:
This results in all user-supplied responses being sent to extras. Accordingly, ExtrasGenerator ignores everything with `overflow: false`, and ExtrasGeneratorV2 builds a more aware question-based data structure.